src/tasks/test-deployment.ts

```diff
@@ -0,0 +1,382 @@
+    weth,
+    usdcPerWeth,
+    chainId,
+  }: CleanArgs,
+  hre: HardhatRuntimeEnvironment,
+) {
+  const { ethers } = hre;
+  const [deployer] = await ethers.getSigners();
+  const salt = id(Date.now().toString());
+  console.log(`Using deployer ${deployer.address}`);
+
+  console.log("Generating user PKs...");
+  const users = Array(userCount)
+    .fill(null)
+    .map((_, i) => {
+      process.stdout.cursorTo(0);
+      process.stdout.write(`${Math.floor((i * 100) / userCount)}%`);
+      return Wallet.fromMnemonic(mnemonic, `m/44'/60'/${i}'/0/0`);
+    });
+  process.stdout.cursorTo(0);
+  const privateKeys: Record<string, string> = {};
+  for (const user of users) {
+    privateKeys[user.address] = user.privateKey;
+  }
+
+  console.log("Generating user claims...");
+  const claims = generateClaims(users.map((user) => user.address));
+
+  console.log("Generating Merkle proofs...");
+  const { merkleRoot, claims: claimsWithProof } = computeProofs(claims);
+
+  // The contracts are deployed from a contract and require that some receiver
+  // addresses are set. All these are created now and are Gnosis Safe.
+  console.log("Deploying administration safes...");
+  const deploySafe: () => Promise<Contract> = async () =>
+    (await deployWithOwners([deployer.address], 1, deployer, hre)).connect(
+      ethers.provider,
+    );
+  const gnosisDao = await deploySafe();
+  const cowDao = await deploySafe();
+  const communityFundsTarget = await deploySafe();
+  const investorFundsTarget = await deploySafe();
+  const teamController = await deploySafe();
+
+  const realTokenDeployParams: RealTokenDeployParams = {
+    totalSupply,
+    cowDao: cowDao.address,
+  };
+
+  const virtualTokenDeployParams: Omit<VirtualTokenDeployParams, "realToken"> =
+    {
+      merkleRoot,
+      communityFundsTarget: communityFundsTarget.address,
+      investorFundsTarget: investorFundsTarget.address,
+      usdcToken: usdc.instance.address,
+      usdcPrice: usdcPerCow,
+      gnoToken: gno.instance.address,
+      gnoPrice: utils
+        .parseUnits("1", gno.decimals)
+        .mul(usdcPerCow)
+        .div(usdcPerGno),
+      wethToken: weth.instance.address,
+      wethPrice: utils
+        .parseUnits("1", weth.decimals)
+        .mul(usdcPerCow)
+        .div(usdcPerWeth),
+      teamController: teamController.address,
+    };
+
+  console.log("Generating deploy transactions...");
+  const {
+    realTokenDeployTransaction,
+    virtualTokenDeployTransaction,
+    realTokenAddress,
+    virtualTokenAddress,
+  } = await prepareSafeDeployment(
+    realTokenDeployParams,
+    virtualTokenDeployParams,
+    MultiSendDeployment.networkAddresses[chainId],
+    ethers,
+    salt,
+  );
+
+  expect(await ethers.provider.getCode(realTokenAddress)).to.equal("0x");
+  expect(await ethers.provider.getCode(virtualTokenAddress)).to.equal("0x");
+
+  console.log("Saving generated data to file...");
+  await fs.rmdir(OUTPUT_FOLDER, { recursive: true });
```

Comment:
> Is this risky? Imagine, I have set the output_folder incorrectly? Maybe just overwritting existing files is safer than deleting the whole dir, as one would only overwrite files with certain names.

The problem with this approach is that the filenames change in the `output/chunks` folder. I could make an `rm` loop in that folder, but I'd say that the risks are similar and this is simpler.

Reply:
I tried to implement the changes I proposed, you can check them out [here](https://github.com/gnosis/cow-token/compare/rinkeby-deployment...rinkeby-deployment-revised-filesystem-access?expand=1). If you think these changes would solve your concern I'll merge them into this branch.